Sources/NIOExtras/LengthFieldBasedFrameDecoder.swift

```diff
@@ -35,11 +85,11 @@ import NIO
 public final class LengthFieldBasedFrameDecoder: ByteToMessageDecoder {
     ///
     /// An enumeration to describe the length of a piece of data in bytes.
-    /// It is contained to lengths that can be converted to integer types.
     ///
     public enum ByteLength {
         case one
         case two
+        case three
```

Comment:
Unfortunately, this is public API breaking 😭

Reply:
I think we should have a new initialiser that takes a new "ByteLength" type that is a) extensible b) contains 24 bits as well. Maybe we can call it `BitLength` and do

```
public struct BitLength {
    private struct Backing {
        case bits8
       case bits16
        case bits24
        ...
    }
    private let bitLength: Backing

    public static oneByte: BitLength = BitLength(bitLength: .bits8)
    public static twoBytes: BitLength = BitLength(bitLength: .bits16)
    public static threeBytes: BitLength = BitLength(bitLength: .bits24)

    public static eithtBits: BitLength = BitLength(bitLength: .bits8)
    public static sixteenBits: BitLength = BitLength(bitLength: .bits16)
    ...
}
```

then we can also deprecate `ByteLength`. @Lukasa wdyt?

Reply:
Oh... I totally forgot about that.
@weissi, your proposed solution sounds good to me 👍